src/api.v2/helpers/pipeline/getPipelineStatus.js

```diff
@@ -0,0 +1,296 @@
+          // a map state can have 0 iterations; e.g. trying to delete previous experiments'
+          // so MapIterationStared won't exist and thus branch won't either
+          if (this.branches[0] !== undefined) {
+            this.completedTasks = this.completedTasks.concat(this.branches[0].completedTasks);
+            this.branches = {};
+            this.branchCount = 0;
+          }
+        }
+      }
+    }
+  }
+
+  if (!events.length) {
+    return [];
+  }
+
+  const main = new Branch(events[0], true);
+  for (let ii = 1; ii < events.length; ii += 1) {
+    const consumer = main.nextConsumer(events[ii]);
+    consumer.consume(events[ii]);
+  }
+
+  let shortestCompleted = null;
+
+  if (main.workingOnBranches()) {
+    const branches = Object.values(main.branches);
+    for (let ii = 0; ii < branches.length; ii += 1) {
+      if (!shortestCompleted || branches[ii].completedTasks.length < shortestCompleted.length) {
+        shortestCompleted = branches[ii].completedTasks;
+      }
+    }
+  }
+
+  shortestCompleted = (shortestCompleted || []).concat(main.completedTasks);
+
+  const shortestCompletedToReport = _.filter(shortestCompleted, (s) => pipelineSteps.includes(s));
+
+  return shortestCompletedToReport || [];
+};
+
+
+/*
+     * Return `completedSteps` of the state machine (SM) associated to the `experimentId`'s pipeline
+     * The code assumes that
+     *  - a step is only considered completed if it has been completed for all iteration of the Map
+     *  - steps are returned in the completion order, and are unique in the returned array
+     */
+const getPipelineStatus = async (experimentId, processName) => {
+  const executions = await new ExperimentExecution().find({ experiment_id: experimentId });
+
+  const pipelineExecution = _.find(executions, { pipelineType: processName });
+
+  // if there aren't ARNs just return NOT_CREATED status
+  if (_.isNil(pipelineExecution)) {
+    return buildNotCreatedStatus(processName);
+  }
+
+  const stepFunctions = new AWS.StepFunctions({
+    region: config.awsRegion,
+  });
+
+  let execution = {};
+  let completedSteps = [];
+  let error = false;
+
+  const { executionArn = null, paramsHash = null } = pipelineExecution;
+
+  try {
+    execution = await stepFunctions.describeExecution({
+      executionArn,
+    }).promise();
+  } catch (e) {
+    // if we get the execution does not exist it means we are using a pulled experiment so
+    // just return a mock sucess status
+    // TODO: state machines in production are deleted after 90 days, return a successful execution
+    // if the execution does not exist in production so the user will not be forced to re-run
+    // the pipeline losing annotations. This will be addressed checking if the
+    // processed files exist in S3 to avoid allowing users to move onwards when the pipeline was not
+    // actually run.
+    if ((e.code === pipelineConstants.EXECUTION_DOES_NOT_EXIST)
```

Comment:
instead of doing this, it would be great to save the result of 
```
 execution = await stepFunctions.describeExecution({
      executionArn,
    }).promise();
```
in the database so that when the data is removed 90 days after from aws we can query it.